mux.go

```diff
@@ -454,7 +455,13 @@ func (mx *Mux) updateSubRoutes(fn func(subMux *Mux)) {
 
 // methodNotAllowedHandler is a helper function to respond with a 405,
 // method not allowed.
-func methodNotAllowedHandler(w http.ResponseWriter, r *http.Request) {
-	w.WriteHeader(405)
-	w.Write(nil)
+func methodNotAllowedHandler(allowedMethods ...string) http.HandlerFunc {
+	fn := func(w http.ResponseWriter, r *http.Request) {
+		// sort them so they always appear in the same order
+		sort.Strings(allowedMethods)
```

Comment:
how about instead of sorting we use the natural order as defined by `methodMap` ? would produce better performance for methods that return 405, without requiring a string sort on the invocation

Reply:
done, thanks for the review!